internal/gengapic/paging.go

```diff
@@ -138,6 +138,15 @@ func (g *generator) pagingField(m *descriptor.MethodDescriptorProto) (*descripto
 			hasNextToken = true
 		}
 		if f.GetLabel() == descriptor.FieldDescriptorProto_LABEL_REPEATED {
+			if f.GetType() == descriptor.FieldDescriptorProto_TYPE_MESSAGE {
```

Comment:
It seems like `diregapicPagingField()` covers a superset of functionality as `pagingField()` and similarly for tests, so there's some duplicated logic. Would it make sense to simply have one function with a flag `allowDIREGAPIC bool`?

Reply:
I think it would make sense to do that. This brings us back to the question of having an explicit diregapic flag, though. I'm in favor of not having a flag and instead treating it like diregapic pagination if it passes the predicate. The two differences are the name variant of page size field, i.e. `max_results`, and maps.

Reply:
Well, the function could have a parameter `allowDIREGAPIC bool`, and we can always set it to `true` at calling sites for now. If/when we introduce a flag for DIREGAPIC, that value will be taken from the flag. 

- Advantage: it marks and flags the DIREGAPIC-specific code right now, while it's fresh in our mind. I t also allows us to test the code with the parameter set to `false` right away.

- Disadvantage: until we do have a top-level flag, all the calling sites will have an extra parameter `(.., true)` essentially. (Though we should make it into a constant for comprehensibility: `(..., allowDIREGAPIC)`

Actually, here's an idea to get rid of that disadvantage: have `allowDIREGAPIC` be a (package?) var referenced *inside* the function(s) that have DIREGAPIC-specific code, and NOT a function param. That way we can still flag DIREGAPIC-specific sections of code, and because it's a test, we can set it to false and test non-DIREGAPIC flows (careful with parallel tests!). It only becomes a function param instead of a package var when we do have a DIREGAPIC flag to pass.

WDYT?

Reply:
> have allowDIREGAPIC be a (package?) var referenced inside the function(s) that have DIREGAPIC-specific code

Maybe it would be cleaner to add a new field in the `generator.options` struct that we use to carry both flags and other config at the generator-level. And if there is ever a plugin option added, it just gets wired up to that.

Reply:
+1, even better! This will allow the tests to be modular without much worry, while preserving all the other benefits.

Reply:
Not dropping the diregapic option, just holding off on this for a little bit. Will post code containing it soon.

Reply:
OK. I think having the bool be always be set in the struct for now make sense, and having code here and in tests use it. And like Noah says, we can wire up to any eventual plugin option later.